### ISSUE TITLE
"AnsibleUndefinedVariable: dict object has no element True" from task "Setup powerlevel10k" in "diodonfrost.p10k/tasks/p10k-configure.yml"

### ISSUE THREAD

**Issue by loungingPrimate** [diodonfrost/ansible-role-p10k#2]
##### SUMMARY
Hi diodonfrost,
While trying to use your role I seem to have hit a wall.
Every time I try to do a play I cannot get past the task "Setup powerlevel10k" in "diodonfrost.p10k/tasks/p10k-configure.yml".
I have gotten a similar error message to this one:
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: dict object has no element True
fatal: [ubu2]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: dict object has no element True"}

This happened before I tried to test
changing the become user like so:
become_user: "{{ ansible_user }}"

(I only need that one user.)

I've found two similar problems while searching:
https://github.com/ansible/ansible/issues/24219
https://stackoverflow.com/questions/65215109/jinja2-failing-to-retrieve-value-of-a-key-in-a-dictionary

Both seem to point to the jinja file(in my case: diodonfrost.p10k/templates/p10k-rainbow.zsh.j2
I looked for any occurence of "True" in the role. Outside of molecule there is only one line in diodonfrost.p10k/tasks/p10k-install.yml
It is for a default value of zsh_plugin which I have not changed.

##### STEPS TO REPRODUCE

change:
- name: Setup powerlevel10k
  template:
    src: "p10k-{{ p10k_style[p10k_prompt_style] }}.zsh.j2"
    dest: '~/.p10k.zsh'
    mode: '0644'
  become: yes
  become_user: "{{ item }}"
  with_items: "{{ p10k_users }}"
to:
- name: Setup powerlevel10k
  template:
    src: "p10k-{{ p10k_style[p10k_prompt_style] }}.zsh.j2"
    dest: '~/.p10k.zsh'
    mode: '0644'
  become: yes
  become_user: "{{ ansible_user }}"

```
---
- name: Setup ssh
  hosts: all
  become: true
- name: Caching facts and setting default values
  import_playbook: playbooks/facts.yml
  tags:
    - always
    - facts
- name: Setup ssh
  import_playbook: playbooks/test.yml
  tags:
    - always
    - ssh
...
```
```
---
- name: Caching facts
  hosts: all
  any_errors_fatal: true
  max_fail_percentage: 0
  gather_facts: false
  vars_files:

  tasks:
    - name: "Gather all facts"
      ansible.builtin.setup:
        gather_subset:
          - 'all'
...
```
```
---
- name: Test
  hosts: all
  roles:
    - role: diodonfrost.p10k
      p10k_prompt_style: rainbow
      p10k_prompt_height: one-line
      p10k_prompt_flow: fluent
      p10k_transient_prompt: yes
...
```

ansible-playbook -i inventory.yml m.yml -K -vvv
##### EXPECTED RESULTS
A running and completing play.

##### ACTUAL RESULTS
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/ansible/template/__init__.py", line 1139, in do_template
    res = j2_concat(rf)
  File "<template>", line 99, in root
  File "/usr/lib/python3.9/site-packages/jinja2/runtime.py", line 903, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: dict object has no element True

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/ansible/plugins/action/template.py", line 150, in run
    resultant = templar.do_template(template_data, preserve_trailing_newlines=True, escape_backslashes=False)
  File "/usr/lib/python3.9/site-packages/ansible/template/__init__.py", line 1176, in do_template
    raise AnsibleUndefinedVariable(e)
ansible.errors.AnsibleUndefinedVariable: dict object has no element True
fatal: [pi2]: FAILED! => {
    "changed": false,
    "msg": "AnsibleUndefinedVariable: dict object has no element True"
}

##### ENVIRONMENT

Targeted are three virtual ubuntus(as a lab): 20.04 LTS
ansible [core 2.12.0]
jinja  version = 3.0.3
running OS: Manjaro
Python version = 3.9.7

**Comment by Serializator** [comment]
**TLDR;** quote the value `yes` so that it becomes `"yes"` for `p10k_transient_prompt`.
```yaml
- role: diodonfrost.p10k
  p10k_prompt_style: rainbow
  p10k_prompt_height: one-line
  p10k_prompt_flow: fluent
  p10k_transient_prompt: "yes"
```

---

I ran into the same problem and decided to look into it, even though the issue is quite old.

The **unquoted** value of `yes` is interpreted as a boolean, `True` in this case. This directly links to the exception.
> AnsibleUndefinedVariable: dict object has no element True

The template is trying to get the value of element `True` in a dictionary, which does not have an element of `True` in it, causing the exception in question.

The only noticeable variable in this case is `p10k_transient_prompt` which accepts either `yes` or `no` as its value. In the templates the value for this variable (`p10k_transient_prompt`) is used as an element in a dictionary.

https://github.com/diodonfrost/ansible-role-p10k/blob/master/templates/p10k-classic.zsh.j2#L1626

`{{ transient_config[p10k_transient_prompt] }}`

If you keep in mind that the **unquoted** value of `yes` is interpreted as a boolean (`True`) then the above is translated into:
`{{ transient_config[True] }}`

Below is what the `transient_config` dictionary looks like.

https://github.com/diodonfrost/ansible-role-p10k/blob/6a0944ab8b109eb863881a37407a6addc1efe1ef/vars/main.yml#L128-L130

Keeping all of this in mind I looked for other mappings with `yes` or `no` in the aforementioned YAML and there are two other mappings.

https://github.com/diodonfrost/ansible-role-p10k/blob/6a0944ab8b109eb863881a37407a6addc1efe1ef/vars/main.yml#L63-L66

https://github.com/diodonfrost/ansible-role-p10k/blob/6a0944ab8b109eb863881a37407a6addc1efe1ef/vars/main.yml#L141-L151

The noticeable thing is that for `time_config` the mapping of `no` is **not** quoted and for `frame_config` it **is** quoted. If we were to configure `p10k_prompt_time` as `no` there wouldn't be a problem. `no` would be interpreted as `False` both as the value for `p10k_prompt_time` and for the mapping in `frame_config`.
